app/models/user.rb

```diff
@@ -630,6 +628,21 @@ def valid_otp?(otp)
     totp.verify(otp, drift_behind: 10)
   end
 
+  def ensure_2fa_token!
```

Comment:
Hmm, I think this naming `ensure.. ` + an `if condition` does not make any sense because we are regenerating this token on each turn on/off. 

I would rename it to something like (re)assign_otp_code. Do you have a better idea?